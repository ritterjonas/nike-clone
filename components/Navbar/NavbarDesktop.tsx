import Image from 'next/image';
import {
  InputSearchContainer,
  InputText,
  MenuItem,
  MenuItemLink,
  NavbarComponentDesktop,
  NavbarAccount,
  NavbarAccountImages,
  NavbarAccountItems,
  NavbarAccountItem,
  NavbarAccountSeparator,
  NavbarIcon,
  NavbarIcons,
  NavbarMenus,
} from './NavbarDesktop.styles';

export default function NavbarDesktop() {
  return (
    <>
      <NavbarAccount>
        <NavbarAccountImages>
          <Image
            src='/images/logo_jordan.svg'
            alt='Nike Logo'
            width={24}
            height={24}
            data-testid='jordan-logo-desktop'
          />
          <Image
            src='/images/logo_snkrs.svg'
            alt='Nike Logo'
            width={48}
            height={24}
            data-testid='snkrs-logo-desktop'
          />
        </NavbarAccountImages>
        <NavbarAccountItems>
          <NavbarAccountItem href='#'>Ajuda</NavbarAccountItem>
          <NavbarAccountSeparator />
          <NavbarAccountItem href='#'>Acompanhe seu pedido</NavbarAccountItem>
          <NavbarAccountSeparator />
          <NavbarAccountItem href='#'>Junte-se a nós</NavbarAccountItem>
          <NavbarAccountSeparator />
          <NavbarAccountItem href='#'>Entrar</NavbarAccountItem>
        </NavbarAccountItems>
      </NavbarAccount>
      <NavbarComponentDesktop data-testid='navbar-desktop'>
        <Image
          src='/images/logo_nike.svg'
          alt='Nike Logo'
          width={40}
          height={15}
          data-testid='nike-logo-desktop'
        />

        <NavbarMenus>
          <MenuItem>
            <MenuItemLink href='#'>Lançamento</MenuItemLink>
          </MenuItem>
          <MenuItem>
            <MenuItemLink href='#'>Masculino</MenuItemLink>
          </MenuItem>
          <MenuItem>
            <MenuItemLink href='#'>Feminino</MenuItemLink>
          </MenuItem>
          <MenuItem>
            <MenuItemLink href='#'>Infantil</MenuItemLink>
          </MenuItem>
          <MenuItem>
            <MenuItemLink href='#'>Ofertas</MenuItemLink>
          </MenuItem>
          <MenuItem>
            <MenuItemLink href='#'>SNKRS</MenuItemLink>
          </MenuItem>
        </NavbarMenus>

        <NavbarIcons>
          <InputSearchContainer>
            <Image
              src='/images/icon_lupa.svg'
              alt='Search icon'
              width={12}
              height={12}
            />
            <InputText type='text' placeholder='Buscar' />
          </InputSearchContainer>

          <NavbarIcon href='#'>
            <Image
              src='/images/icon_sacola.svg'
              alt='Bag Icon'
              width={20}
              height={20}
            />
          </NavbarIcon>
        </NavbarIcons>
      </NavbarComponentDesktop>
    </>
  );
}
