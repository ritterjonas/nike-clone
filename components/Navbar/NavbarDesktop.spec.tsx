import React from 'react';
import { render } from '@testing-library/react';
import NavbarDesktop from './NavbarDesktop';

describe('<NavbarDesktop />', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<NavbarDesktop />);
    expect(asFragment()).toMatchSnapshot();
  });
});
