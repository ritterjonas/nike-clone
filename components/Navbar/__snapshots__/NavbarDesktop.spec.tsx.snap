// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarDesktop /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdbead5f-1 Tapag"
  >
    <div
      class="sc-bdbead5f-2 efHBR"
    >
      <img
        alt="Nike Logo"
        data-nimg="1"
        data-testid="jordan-logo-desktop"
        decoding="async"
        height="24"
        loading="lazy"
        src="/images/logo_jordan.svg"
        style="color: transparent;"
        width="24"
      />
      <img
        alt="Nike Logo"
        data-nimg="1"
        data-testid="snkrs-logo-desktop"
        decoding="async"
        height="24"
        loading="lazy"
        src="/images/logo_snkrs.svg"
        style="color: transparent;"
        width="48"
      />
    </div>
    <div
      class="sc-bdbead5f-3 gAQxAM"
    >
      <a
        class="sc-bdbead5f-4 kUVdiI"
        href="#"
      >
        Ajuda
      </a>
      <div
        class="sc-bdbead5f-5 byFZNl"
      />
      <a
        class="sc-bdbead5f-4 kUVdiI"
        href="#"
      >
        Acompanhe seu pedido
      </a>
      <div
        class="sc-bdbead5f-5 byFZNl"
      />
      <a
        class="sc-bdbead5f-4 kUVdiI"
        href="#"
      >
        Junte-se a nós
      </a>
      <div
        class="sc-bdbead5f-5 byFZNl"
      />
      <a
        class="sc-bdbead5f-4 kUVdiI"
        href="#"
      >
        Entrar
      </a>
    </div>
  </div>
  <div
    class="sc-bdbead5f-0 gEbleI"
    data-testid="navbar-desktop"
  >
    <img
      alt="Nike Logo"
      data-nimg="1"
      data-testid="nike-logo-desktop"
      decoding="async"
      height="15"
      loading="lazy"
      src="/images/logo_nike.svg"
      style="color: transparent;"
      width="40"
    />
    <div
      class="sc-bdbead5f-8 ghfAYz"
    >
      <div
        class="sc-bdbead5f-9 lmiXWG"
      >
        <a
          class="sc-bdbead5f-10 bSvaFL"
          href="#"
        >
          Lançamento
        </a>
      </div>
      <div
        class="sc-bdbead5f-9 lmiXWG"
      >
        <a
          class="sc-bdbead5f-10 bSvaFL"
          href="#"
        >
          Masculino
        </a>
      </div>
      <div
        class="sc-bdbead5f-9 lmiXWG"
      >
        <a
          class="sc-bdbead5f-10 bSvaFL"
          href="#"
        >
          Feminino
        </a>
      </div>
      <div
        class="sc-bdbead5f-9 lmiXWG"
      >
        <a
          class="sc-bdbead5f-10 bSvaFL"
          href="#"
        >
          Infantil
        </a>
      </div>
      <div
        class="sc-bdbead5f-9 lmiXWG"
      >
        <a
          class="sc-bdbead5f-10 bSvaFL"
          href="#"
        >
          Ofertas
        </a>
      </div>
      <div
        class="sc-bdbead5f-9 lmiXWG"
      >
        <a
          class="sc-bdbead5f-10 bSvaFL"
          href="#"
        >
          SNKRS
        </a>
      </div>
    </div>
    <div
      class="sc-bdbead5f-6 ieLyPj"
    >
      <div
        class="sc-bdbead5f-11 cUKETF"
      >
        <img
          alt="Search icon"
          data-nimg="1"
          decoding="async"
          height="12"
          loading="lazy"
          src="/images/icon_lupa.svg"
          style="color: transparent;"
          width="12"
        />
        <input
          class="sc-bdbead5f-12 fkvgyl"
          placeholder="Buscar"
          type="text"
        />
      </div>
      <a
        class="sc-bdbead5f-7 kSunjU"
        href="#"
      >
        <img
          alt="Bag Icon"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/images/icon_sacola.svg"
          style="color: transparent;"
          width="20"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
